### PR TITLE
5.7.36 codership mysql wsrep 5.7.35 25.27

### PR DESCRIPTION
**Pull request by nitendra-bhosle**

### CODE REVIEW

**Comment by nitendra-bhosle** [comment]
Jenkins : https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/262/

**Comment by nitendra-bhosle** [comment]
Latest Jenkins : https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/264/

File: scripts/wsrep_sst_rsync.sh [hunk 1]
```diff
@@ -178,6 +178,21 @@ FILTER=(-f '- /lost+found'
 if [ "$WSREP_SST_OPT_ROLE" = "donor" ]
 then
 
+||||||| merged common ancestors
```
Comment (venkatesh-prasad-v) [inline scripts/wsrep_sst_rsync.sh]: Version Control Marker will cause rsync tests to fail. Please remove this.


>2022-01-19T07:14:07.489217Z 0 [Note] WSREP: Initiating SST/IST transfer on JOINER side (wsrep_sst_rsync --role 'joiner' --address '127.0.0.1:13009' --datadir '/home/venki/work/pxc/review/nitendra/bld/mysql-test/var/mysqld.2/data/' --defaults-file '/home/venki/work/pxc/review/nitendra/bld/mysql-test/var/my.cnf' --defaults-group-suffix '.2' --parent '39359' --mysqld-version '5.7.36-39'   '' ) 
/home/venki/work/pxc/review/nitendra/bld/scripts/wsrep_sst_rsync: line 181: syntax error near unexpected token `||'
>2022-01-19T07:14:07.661565Z 0 [ERROR] WSREP: Failed to read 'ready <addr>' from: wsrep_sst_rsync --role 'joiner' --address '127.0.0.1:13009' --datadir '/home/venki/work/pxc/review/nitendra/bld/mysql-test/var/mysqld.2/data/' --defaults-file '/home/venki/work/pxc/review/nitendra/bld/mysql-test/var/my.cnf' --defaults-group-suffix '.2' --parent '39359' --mysqld-version '5.7.36-39'   '' 
	Read: '(null)'

File: scripts/wsrep_sst_rsync.sh [hunk 2]
```diff
@@ -178,6 +178,21 @@ FILTER=(-f '- /lost+found'
...
+${CAFILE_OPT}
+foreground = yes
+pid = $STUNNEL_PID
+debug = warning
+client = yes
+connect = ${WSREP_SST_OPT_HOST_UNESCAPED}:${WSREP_SST_OPT_PORT}
+TIMEOUTclose = 0
+${VERIFY_OPT}
+EOF
+#connect = ${WSREP_SST_OPT_ADDR%/*}
```
Comment (venkatesh-prasad-v) [inline scripts/wsrep_sst_rsync.sh]: We reverted Codership's stunnel encryption in rsync by commit 998da9b5d. So this block can completely be removed.

File: sql/log_event.cc [hunk 3]
```diff
@@ -10500,6 +10500,30 @@ int Rows_log_event::handle_idempotent_and_ignored_errors(Relay_log_info const *r
...
     bool ignored_error= (idempotent_error == 0 ?
                          ignored_error_code(actual_error) : 0);
+#if defined(WITH_WSREP)
+    /*
+     * Work around mysql Bug#80821. With cascading deletes, when both parent and
+     * child tables are involved in a multi table query, a delete rows event for
+     * the child may be added to the binlog after the parent (cascading) delete.
+     * This will cause the delete to fail on slaves. A new wsrep_mode value has
+     * been added for ignoring such delete errors.
+     */
```
Comment (venkatesh-prasad-v) [inline sql/log_event.cc:10510]: IIUC, [Bug#80821](https://bugs.mysql.com/bug.php?id=80821) is already fixed in PS and PXC. So I'm not sure if this is really necessary.

@dutow Can you please share your opinions on this?

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]

**Comment by nitendra-bhosle** [comment]
New Jenkins : https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/267/

**Comment by nitendra-bhosle** [comment]
Jenkins: https://pxc.cd.percona.com/view/PXC%205.7/job/pxc-5.7-param/268/

File: sql/log_event.cc [hunk 4]
```diff
@@ -10500,6 +10500,30 @@ int Rows_log_event::handle_idempotent_and_ignored_errors(Relay_log_info const *r
                            (rbr_exec_mode == RBR_EXEC_MODE_IDEMPOTENT));
     bool ignored_error= (idempotent_error == 0 ?
                          ignored_error_code(actual_error) : 0);
+#if defined(WITH_WSREP)
```
Comment (venkatesh-prasad-v) [inline sql/log_event.cc]: Please change it to `#ifdef WITH_WSREP` in order to make it work with our `check_src` tool.

File: sql/sql_parse.cc [hunk 5]
```diff
@@ -6379,13 +6380,7 @@ case SQLCOM_PREPARE:
   mysql_mutex_lock(&thd->LOCK_wsrep_thd);
   if (thd->wsrep_conflict_state != REPLAYED)
   {
-  /* Hold this lock only for duration of wsrep_conflict_state state check.
-  Release it before entering unit-cleanup. Unit-cleanup will cause
-  external_unlock at InnoDB level that will need transaction lock.
-  This could conflict if this thread has conflicting lock with
-  other high-priority thread that needs transaction lock and
-  LOCK_wsrep_thd too. */
```
Comment (venkatesh-prasad-v) [inline sql/sql_parse.cc:6387]: Please revert this change (bring back the comment).

**Review by venkatesh-prasad-v — APPROVED** [review]
lgtm

**Review by venkatesh-prasad-v — APPROVED** [review]

**Pull request merged**